File: frontend/app/src/interfaces/Fornecedor/FornecedorEmpResponseDTO.ts
import { EmpresaResponseDTO } from "../Empresa/EmpresaResponseDTO";

export default interface FornecedorEmpResponseDTO {
   id: string,
   cnpj_cpf: string,
   rg: string,
   nascimento: Date,
   nome: string,
   email: string,
   cep: string,
   e_pf: boolean,
   empresas: EmpresaResponseDTO[]
}

File: backend/test/src/main/java/com/accenture/test/adapter/output/fornecedor/FornecedorEmpResponseDTO.java
package com.accenture.test.adapter.output.fornecedor;

import com.accenture.test.adapter.output.empresa.EmpresaResponseDTO;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record FornecedorEmpResponseDTO(
        UUID id,
        String cnpj_cpf,
        String rg,
        LocalDate nascimento,
        String nome,
        String email,
        String cep,
        boolean e_pf,
        List<EmpresaResponseDTO> empresas
) {}
